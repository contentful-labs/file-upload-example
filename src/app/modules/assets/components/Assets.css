@import '../../../assets/styles/variables.css';

.wrapper {}

.dropzone {
  &:before {
    content: 'Drag & drop files anywhere to upload';
    display: inline-block;
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-50%);
    padding: 0.5em;
    border: 1px solid black;
    border-top: none;
    border-bottom-right-radius: 4px;
    border-bottom-left-radius: 4px;
    font-size: 0.8em;
    opacity: 0.6;
  }
}

.dropzoneActive {
  background: #ddd;
}
