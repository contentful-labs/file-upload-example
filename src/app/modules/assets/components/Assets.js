import { h, Component } from 'preact'
import proptypes from 'proptypes'
import Dropzone from 'react-dropzone'

import styles from './Assets.css'

export default class Assets extends Component {
  static propTypes = {
    assets: proptypes.array.isRequired,
    uploadFiles: proptypes.func.isRequired
  }
  constructor (props) {
    super(props)
    this._onDrop = this._onDrop.bind(this)
  }
  _onDrop (files) {
    this.props.uploadFiles(files)
  }
  render ({ assets }) {
    const assetsList = assets.map((asset) => {
      const localeFile = Object.keys(asset.fields.file)[0]
      const localeTitle = Object.keys(asset.fields.title)[0]
      const title = asset.fields.title[localeTitle]
      const url = asset.fields.file[localeFile].url
      return (
        <li key={asset.sys.id}>
          <h3>{title}</h3>
          <img src={url} alt={title} />
        </li>
      )
    })
    return (
      <div>
        <Dropzone
          disableClick
          onDrop={this._onDrop}
          className={styles.dropzone}
          activeClassName={styles.dropzoneActive} >
          <h1>Your assets:</h1>
          <ul className={styles.wrapper}>
            {assetsList}
          </ul>
        </Dropzone>
      </div>
    )
  }
}
