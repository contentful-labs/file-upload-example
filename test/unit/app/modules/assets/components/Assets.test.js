import { h } from 'preact'
import render from 'preact-render-to-string'

import Assets from '../../../../../../src/app/modules/assets/components/Assets'

test('Assets - Empty', () => {
  const uploadMock = jest.fn()
  const html = render(<Assets assets={[]} uploadFiles={uploadMock} />)
  expect(html).toMatchSnapshot()
  expect(uploadMock.mock.calls).toHaveLength(0)
})

test('Assets - Filled', () => {
  const uploadMock = jest.fn()
  const assets = [
    {
      sys: {
        id: 'invalid asset - should be filtered'
      },
      fields: {}
    },
    {
      sys: {
        id: 'valid',
        createdAt: new Date()
      },
      fields: {
        title: {
          'en-US': 'totally valid asset'
        },
        file: {
          'en-US': {
            fileName: 'awesome-image.jpg',
            url: 'http://example.com/awesome-image.jpg',
            contentType: 'image/awesome-format'
          }
        }
      }
    }
  ]
  const html = render(<Assets assets={assets} uploadFiles={uploadMock} />)
  expect(html).toMatchSnapshot()
  expect(uploadMock.mock.calls).toHaveLength(0)
})
