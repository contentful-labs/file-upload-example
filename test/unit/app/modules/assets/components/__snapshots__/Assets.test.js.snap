// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets - Empty 1`] = `
<div>
  <div class="dropzone">
    <a class="changeSpaceLink" href="/">Change space</a>
    <h1>Your assets:</h1>
    <div class="assets"></div>
    <input type="file" style="display: none;" multiple autoComplete="off"
    />
  </div>
</div>
`;

exports[`Assets - Filled 1`] = `
<div>
  <div class="dropzone">
    <a class="changeSpaceLink" href="/">Change space</a>
    <h1>Your assets:</h1>
    <div class="assets">
      <div class="asset">
        <div class="label">
          <div class="labelContent">
            <div class="title">totally valid asset</div>
            <div class="created">created less than 5 seconds ago</div>
          </div>
        </div>
        <img src="http://example.com/awesome-image.jpg?w=800&amp;h=600&amp;q=80&amp;fm=jpg&amp;fl=progressive&amp;fit=fill"
        alt="totally valid asset" />
      </div>
    </div>
    <input type="file" style="display: none;" multiple autoComplete="off"
    />
  </div>
</div>
`;
